subsys/nfc/lib/Kconfig

```diff
@@ -21,6 +21,94 @@ config NFCT_IRQ_PRIORITY
 	  Sets NFC interrupt priority.
 	  Levels are from 0 (highest priority) to 6 (lowest priority)
 
+config NFC_THREAD_CALLBACK
+	bool "Thread context for NFC callbacks"
+	select RING_BUFFER
+	default y
+	help
+	  Use this option to decouple the user NFC callback
+	  from the NFC interrupt context. When enabled, the user
+	  callback is called from a thread (System Workqueue
+	  or a dedicated thread depending on the CONFIG_NFC_OWN_THREAD
+	  setting) instead of being called from the interrupt context.
+	  Using this option ensures robust NFC communication
+	  (deterministic time of the interrupt handling) and allows
+	  deferred callback execution, while it results in higher memory
+	  footprint and CPU load.
+
+if NFC_THREAD_CALLBACK
+
+config NFC_RING_EXP
+	int "Exponent of ring buffer size"
+	default 9
+	help
+	  Power of 2 exponent of ring buffer size.
+	  The specified size is calculated as 2^n words.
+
+config NFC_MAX_ITEM_SIZE
+	int "Max ring buffer item size in words"
+	default 68
+
+config NFC_OWN_THREAD
+	bool "Call NFC callback from its own thread"
+	default n
+
+if NFC_OWN_THREAD
+
+config NFC_THREAD_STACK_SIZE
+	int "Size of NFC thread stack"
+	default 256
+
+config NFC_THREAD_PRIORITY
+	int "Priority of NFC callback thread"
+	default -1
+
+endif# NFC_OWN_THREAD
+
+config NFC_ZERO_LATENCY_IRQ
+	bool "Use Zero Latency IRQ for NFC"
+	default y
+	select ZERO_LATENCY_IRQS
+	help
+	  Using this option ensures fast NFC interrupt
+	  execution on the cost of using additional interrupt (SWI).
+
+if NFC_ZERO_LATENCY_IRQ
+
+choice NFC_SWI
+	default NFC_SWI0
```

Comment:
We might have a collision here with different radio protocols like gazell, esb, mpsl. Could you check that ?

Reply:
Good point! @dchat-nordic please check which SWI instances are used in different protocols. At the moment, no allocator is used and on different targets different instances are used: `nrf52dk_nrf52832` vs `nrf5340dk_nrf5340_cpuapp`. It is important to ensure coexistence with BLE and Thread protocols.